### PR TITLE
Add doctests for Semigroup, Group, and Monoid

### PR DESCRIPTION
**Pull request by DennisVDB**
PR for part of #2479.

Adds doctests for the `Semigroup`, `Group`, and `Monoid` typeclasses.

### CODE REVIEW

**Comment by DennisVDB** [comment]
`sbt validate` is failing with:

```
error] /Users/dennis/Documents/Programming/scala/cats/kernel/.jvm/target/scala-2.12/src_managed/test/cats/kernel/GroupDoctest.scala:4:19: object scalacheck is not a member of package org
[error] import _root_.org.scalacheck.Prop._
[error]                   ^
[error] /Users/dennis/Documents/Programming/scala/cats/kernel/.jvm/target/scala-2.12/src_managed/test/cats/kernel/GroupDoctest.scala:7:24: object scalacheck is not a member of package org
[error]     extends _root_.org.scalacheck.Properties("Group.scala") {
[error]                        ^
[error] /Users/dennis/Documents/Programming/scala/cats/kernel/.jvm/target/scala-2.12/src_managed/test/cats/kernel/GroupDoctest.scala:7:46: no arguments allowed for nullary constructor Object: ()Object
[error]     extends _root_.org.scalacheck.Properties("Group.scala") {
[error]                                              ^
[error] /Users/dennis/Documents/Programming/scala/cats/kernel/.jvm/target/scala-2.12/src_managed/test/cats/kernel/GroupDoctest.scala:15:3: not found: value include
[error]   include(new _root_.org.scalacheck.Properties("Group.scala:10: inverse") {
...
```

**Comment by kailuowang** [comment]
probably need to add this setting to the kernel settings
```
.settings(libraryDependencies += "org.scalacheck" %%% "scalacheck" % scalaCheckVersion(scalaVersion.value) % "test")

```

File: kernel/src/main/scala/cats/kernel/Monoid.scala [hunk 1]
```diff
@@ -31,6 +64,17 @@ trait Monoid[@sp(Int, Long, Float, Double) A] extends Any with Semigroup[A] {
 
   /**
    * Given a sequence of `as`, sum them using the monoid and return the total.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   *
+   * scala> Monoid[String].combineAllOption(List("One ", "Two ", "Three"))
```
Comment (LukaJCB) [inline kernel/src/main/scala/cats/kernel/Monoid.scala]: Don't you mean `combineAll` here?

**Comment by kailuowang** [comment]
I might missed something but I don't see build.sbt updated

**Comment by DennisVDB** [comment]
You didn't miss it. 

I now pushed the updated `build.sbt` with the fix you suggested but it still fails the same way as before. It seems that the code generated for the doctest  doesn't compile.

The generated code: `/cats/kernel/.jvm/target/scala-2.12/src_managed/test/cats/kernel/*Doctest.scala`

Travis failure: https://travis-ci.org/typelevel/cats/jobs/432625136
Local failure: https://gist.github.com/DennisVDB/1fa934962f2f192af80731b1336adb1d

File: kernel/src/main/scala/cats/kernel/Group.scala [hunk 2]
```diff
@@ -11,13 +11,29 @@ trait Group[@sp(Int, Long, Float, Double) A] extends Any with Monoid[A] {
    * Find the inverse of `a`.
    *
    * `combine(a, inverse(a))` = `combine(inverse(a), a)` = `empty`.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
```
Comment (larsrh) [inline kernel/src/main/scala/cats/kernel/Group.scala]: The problem is that `cats.implicits` is only defined in `cats-core`, not in `cats-kernel`, so importing it here won't work.

**Pull request merged**